### DISCUSSION TITLE
Remove map property from vacuum entity

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#939] [Entity Models] [answered]
### Suggestion:
Remove the `map` property from the vacuum entity.

### Rationale:
Instead of exposing the vacuum cleaner's map status as a vacuum entity state attribute, the map should be exposed as a separate `image` entity

#### Potential drawback:
If the map is shown by a frontend card, it may not be possible to know that the entity represents the vacuum cleaner's map

**Comment by rytilahti** [comment]
When I saw the proposal for extracting the battery entity, I was wondering about the same potential drawback. Maybe the `device_class`  information could be used by the frontend to pick the right entity to show? Having something like this would also be necessary if fan speeds & fan speed presets are to be moved to their own entities at some point.

**Reply by emontnemery** [reply]
In case of replacing the battery property with a sensor, the already existing sensor device class "battery" should be enough.

In this case, we need some way for an image entity to hint that it represents a vacuum cleaner's interior map. Adding a device class to image for that purpose doesn't seem quite right.

**Reply by gjohansson-ST** [reply]
Isn't this pretty obvious though if it belongs to the same device?

**Comment by frenck** [comment, marked as answer]
This is not used by Core, so removing it here seems to be a good solution.
